Build errors when `--out-dir` includes dot in path segment

### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [ ] I have tried the `npm run clear` or `yarn clear` command.
- [ ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [ ] I have read the console error message carefully (if applicable).

### Description

When calling `npm run build -- --out-dir`, if the directory contains a path segment with a dot `.`, the build fails with errors.

For example, the directory `build/product/6.0/` causes errors.


### Reproducible demo

_No response_

### Steps to reproduce

Tested using git bash on Windows 10.

1. Create a new docusaurus site:
    ```
    npx create-docusaurus@latest my-website classic
    ```
1. Go to `my-website` and run:
    ```
    npm run build -- --out-dir "build/product/6.0/"
    ```




### Expected behavior

The site builds correctly at the directory `build/product/6.0/`, relative to the docusaurus project.

### Actual behavior


The results:

```
$ npm run build -- --out-dir "build/product/6.0/"

> my-website@0.0.0 build
> docusaurus build --out-dir build/product/6.0/

[INFO] [en] Creating an optimized production build...

✔ Client


● Server █████████████████████████ sealing (92%) asset processing
 RealContentHashPlugin

[ERROR] Error: EISDIR: illegal operation on a directory, open 'C:\temp\my-website\build\product\6.0'
[ERROR] Unable to build website for locale en.
[ERROR] Error: EISDIR: illegal operation on a directory, open 'C:\temp\my-website\build\product\6.0'
[INFO] Docusaurus version: 2.1.0
Node version: v18.9.0
``` 

The build output is partially successful, but is missing the `docs` and `img` directories:

![image](https://user-images.githubusercontent.com/1324114/192339154-48b07a7b-a88d-421e-9ea3-3ac603a1c8da.png)

### Your environment

- Public source code:
- Public site URL:
- Docusaurus version used: 2.1.0
- Environment name and version (e.g. Chrome 89, Node.js 16.4): Node .js 18.9.0
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): Windows 10


### Self-service

- [ ] I'd be willing to fix this bug myself.

--- a/packages/docusaurus/src/commands/build.ts
+++ b/packages/docusaurus/src/commands/build.ts
@@ -199,7 +199,11 @@ async function buildLocale({
     serverConfig = merge(serverConfig, {
       plugins: [
         new CopyWebpackPlugin({
-          patterns: staticDirectories.map((dir) => ({from: dir, to: outDir})),
+          patterns: staticDirectories.map((dir) => ({
+            from: dir,
+            to: outDir,
+            toType: 'dir',
+          })),
         }),
       ],
     });

